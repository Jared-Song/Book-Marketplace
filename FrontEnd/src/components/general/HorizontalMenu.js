import { Button, Grid } from "@material-ui/core";
import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  selectedItem: {
    width: "100%",
    backgroundColor: "#EBF0FF",
    borderRadius: 15,
    cursor: 'pointer'
  },
  item: {
    width: "100%",
    cursor: 'pointer'
  },
}));

// menuItem shoud looks like: [{title: String, selected: Boolean!, action: () => void}]
export default function HorizontalMenu({ menuItems }) {
  const classes = useStyles();

  return (
    <Grid
      container
      direction="column"
      justifyContent="flex-start"
      alignItems="flex-start"
      spacing={2}
    >
      {menuItems.map((menuItem) => (
        <Grid
          item
          xs={12}
          key={menuItem.title}
          className={menuItem.selected ? classes.selectedItem : classes.item}
        >
          <Button disabled={menuItem.selected} onClick={menuItem.onClick}>{menuItem.title}</Button>
        </Grid>
      ))}
    </Grid>
  );
}
