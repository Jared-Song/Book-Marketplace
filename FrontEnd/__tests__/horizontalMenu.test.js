import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import HorizontalMenu from "../src/components/general/HorizontalMenu";

it("renders HorizontalMenu correctly", () => {
  const tree = renderer
    .create(
      <HorizontalMenu
        menuItems={[
          {
            title: "Requests",
            selected: false,
            onClick: () => {
              router.push("/admin/requests");
            },
          },
          {
            title: "Books",
            selected: false,
            onClick: () => {
              router.push("/admin/books");
            },
          },
          {
            title: "User Management",
            selected: false,
            onClick: () => {
              router.push("/admin/users");
            },
          },
          {
            title: "Order History",
            selected: false,
            onClick: () => {
              router.push("/admin/orders");
            },
          },
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
