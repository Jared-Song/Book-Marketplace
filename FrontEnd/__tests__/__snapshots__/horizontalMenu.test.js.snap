// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HorizontalMenu correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
>
  <div
    className="makeStyles-root-1"
  >
    <nav
      className="MuiList-root MuiList-padding"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Requests
          </span>
        </div>
      </div>
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Books
          </span>
        </div>
      </div>
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            User Management
          </span>
        </div>
      </div>
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Order History
          </span>
        </div>
      </div>
    </nav>
    <hr
      className="MuiDivider-root"
    />
  </div>
</div>
`;
